lapack/gonum/dgesdc2.go

```diff
@@ -0,0 +1,50 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gonum
+
+import (
+	"math"
+
+	"gonum.org/v1/gonum/blas/blas64"
+)
+
+// Dgesdc2 solves a system of linear equations
+//   A * X = scale* RHS
+// with a general N-by-N matrix A using the LU factorization with
+// complete pivoting computed by Dgetc2.
+func (impl Implementation) Dgesdc2(n int, a []float64, lda int, rhs []float64, ipiv, jpiv []int, scale float64) {
+	const eps = dlamchP
+	var smlnum, _ = impl.Dlabad(dlamchS/eps, 1./(dlamchS/eps))
+	if len(a) < (n-1)*lda+n {
+		panic(shortA)
+	}
+	//Apply permutations IPIV to RHS
+	impl.Dlaswp(1, rhs, lda, 1, n-1, ipiv, 1)
+
+	// solve for L part
+	for i := 0; i < n; i++ {
+		// of course this can be optimized
+		// I'm still struggling with this indexing.
+		for j := i + 1; j < n; j++ {
+			rhs[j] = rhs[j] - a[(i*n)+j]*rhs[i]
+		}
+	}
+	scale = 1.
+	//
+	i := impl.Idamax(n, rhs, 1)
+	if 2*smlnum*math.Abs(rhs[i]) > math.Abs(a[n*n+n]) {
+		temp := 0.5 / math.Abs(rhs[i])
+		blas64.Implementation().Dscal(n, temp, rhs[1:], 1) // what?
```

Comment:
```
bi.Scal(rhs[0], blas64.Vector{N: n, Data: temp, Inc: 1})
```

This is scaling `temp` by the first element of `rhs`. The blas64 API takes a vector type here.

Reply:
Again I'm getting the same problem where `blas64.Scal` has the signature you describe while `bi` has `Dscal` available.

Second observation: `temp` is not an array (or is it??). I'm seeing it the other way around, a part of `rhs` (a single element?) is being scaled by `temp`.  Now I'm looking at it more closely, I think it's scaling `rhs[0:]`. Please confirm.

Reply:
> Again I'm getting the same problem where `blas64.Scal` has the signature you describe while `bi` has `Dscal` available.

Sorry, `bi.Dscal`.

> `temp` is not an array (or is it??). I'm seeing it the other way around, a part of `rhs` (a single element?) is being scaled by temp. Now I'm looking at it more closely, I think it's scaling `rhs[0:]`. Please confirm.

Again, sorry. You are right. `temp` is a scalar and `rhs(1)` is being passed to be scaled.